native/cocos/platform/android/jni/JniCocosEntry.cpp

```diff
@@ -23,10 +23,30 @@
  THE SOFTWARE.
 ****************************************************************************/
 
-#pragma once
+#include <jni.h>
 
-//#include <android/asset_manager.h>
-//#include <android/native_window.h>
-//#include <condition_variable>
-//#include <mutex>
-//#include "base/std/container/string.h"
\ No newline at end of file
+#include "platform/android/AndroidPlatform.h"
+#include "platform/android/FileUtils-android.h"
+#include "platform/java/jni/JniHelper.h"
+#include "game-activity/native_app_glue/android_native_app_glue.h"
+
+
+extern "C" {
+
+void android_main(struct android_app *app) {
+    cc::FileUtilsAndroid::setassetmanager(app->activity->assetManager);
+
+    auto *platform = cc::BasePlatform::getPlatform();
+    auto *androidPlatform = static_cast<cc::AndroidPlatform *>(platform);
+    androidPlatform->setAndroidApp(app);
+    platform->init();
+    platform->loop();
```

Comment:
Is it possible for us to use 'run' as the main entry, and run the loop inside. 'app' can also be passed in via argc and argv

Reply:
This entry is the same as other platforms, whether it is also considered to be placed in the template layer. What if the developer wants to extend the interface of the platform?

Reply:
> This entry is the same as other platforms, whether it is also considered to be placed in the template layer. What if the developer wants to extend the interface of the platform?

Can not move to template layer.  cocos_main will trigger renderer initialize. The old code call cocos_main after ANativeWindow is being set.

Reply:
> Is it possible for us to use 'run' as the main entry, and run the loop inside. 'app' can also be passed in via argc and argv

struct android_app ** can not be force cast to const char**

Reply:
> Is it possible for us to use 'run' as the main entry, and run the loop inside. 'app' can also be passed in via argc and argv

argc's type is const char **, is it a good way to do this.

Reply:
It was char **argv before, but it is const char** argv in IOS. So I changed it to const char** uniformly

Reply:
Is it possible to change it to template function ?

Reply:
Not very good, because the design of this function is to keep the same parameters passed in as main, and it can be used to parse the command line in the platform.
Can also pass in multiple parameters